src/astro_databricks/operators/notebook.py

```diff
@@ -245,14 +245,32 @@ def execute(self, context: Context) -> Any:
         :param context:
         :return:
         """
-        if not (
-            hasattr(self.task_group, "is_databricks")
-            and getattr(self.task_group, "is_databricks")
-        ):
-            self.launch_notebook_job()
-        else:
+        if self.databricks_task_group:
             # if we are in a workflow, we assume there is a metadata from the launch task
             databricks_metadata = DatabricksMetaData(**self.databricks_metadata)
             self.databricks_run_id = databricks_metadata.databricks_run_id
             self.databricks_conn_id = databricks_metadata.databricks_conn_id
+        else:
+            self.launch_notebook_job()
+
         self.monitor_databricks_job()
+
+    @property
+    def databricks_task_group(self) -> DatabricksWorkflowTaskGroup | None:
+        """
+        Traverses up parent TaskGroups until the `is_databricks` flag is found.
+        If found, returns the task group. Otherwise, returns None.
+        """
```

Comment:
Please add a note that explains that at the top level, a task would have a task_group of "None" and how that is the exit condition of the loop. Just to make clear that we're depending on an "airflow-ism" to make this function work.